File: Exercism Codes/JavaScript/custom-set.js
export class CustomSet {
  constructor(set = []) {
    this.set = new Set(set);
  }

  empty() {
    return this.set.size === 0;
  }

  contains(el) {
    return this.set.has(el);
  }

  add(el) {
    return new CustomSet(this.set.add(el));
  }

  subset(other) {
    return this.set.isSubsetOf(other.set);
  }

  disjoint(other) {
    return this.set.isDisjointFrom(other.set);
  }

  eql(other) {
    return this.set.size === other.set.size && [...this.set].every((x) => other.set.has(x));
  }

  union(other) {
    return new CustomSet(this.set.union(other.set));
  }

  intersection(other) {
    return new CustomSet(this.set.intersection(other.set));
  }

  difference(other) {
    return new CustomSet(this.set.difference(other.set));
  }
}

File: Exercism Codes/C#/CustomSet.cs

public class CustomSet
{
    private HashSet<int> values;

    public CustomSet(int[] values = null) => this.values = new HashSet<int>(values ?? []);

    public CustomSet Add(int value) => new(values.Append(value).ToArray());

    public bool Empty() => values.Count == 0;

    public bool Contains(int value) => values.Contains(value);

    public bool Subset(CustomSet right) => values.All(right.Contains);

    public bool Disjoint(CustomSet right) => !values.Any(right.Contains);

    public CustomSet Intersection(CustomSet right) => new(values.Intersect(right.values).ToArray());

    public CustomSet Difference(CustomSet right) => new(values.Except(right.values).ToArray());

    public CustomSet Union(CustomSet right) => new(values.Union(right.values).ToArray());

    public override bool Equals(object? obj) => obj is not CustomSet other ? false : other.values.SetEquals(this.values);

    public override string ToString() => string.Join(", ", values);
}